src/runtime/pkg/oci/utils.go

```diff
@@ -689,7 +689,7 @@ func addHypervisorCPUOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig) e
 	numCPUs := goruntime.NumCPU()
 
 	if err := newAnnotationConfiguration(ocispec, vcAnnotations.DefaultVCPUs).setFloat32WithCheck(func(vcpus float32) error {
-		if vcpus > float32(numCPUs) {
+		if vcpus > float32(numCPUs) && sbConfig.HypervisorType != vc.RemoteHypervisor {
```

Comment:
Sorry Ajay - I'm not sure if this code will be enough - I just gave it as an example. I think it is still executing `newAnnotationConfiguration` which has the same error check. I think it is worth seeing if there is a unit test that can be added to check the logic if possible?

We also need the same logic for the memory annotation